staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -275,7 +282,287 @@ func TestDecode(t *testing.T) {
+			in:   hex("3bffffffffffffffff"), // -18446744073709551616
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				if e.CBORType != "negative integer" || e.GoType != "int64" {
+					t.Errorf("Expected types: negative integer,int64. Got: %#v,%#v", e.CBORType, e.GoType)
+				}
+			}),
+		},
+		{
+			name:          "tag 0 timestamp string",
+			in:            hex("c074323030362d30312d30325431353a30343a30355a"), // 0("2006-01-02T15:04:05Z")
+			want:          "2006-01-02T15:04:05Z",
+			failureReason: "decoding cbor data tagged with 0 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:            "tag 0 with byte string",
+			in:              hex("c043414243"), // 0('ABC')
+			unexportedError: "cbor: tag number 0 must be followed by text string, got byte string",
+		},
+		{
+			name:            "tag 0 with invalid text string",
+			in:              hex("c06474657874"), // 0("text")
+			unexportedError: "cbor: cannot set text for time.Time: parsing time",
+		},
+		{
+			name:          "tag 1 timestamp unsigned integer",
+			in:            hex("c11a43b940e5"), // 1(1136214245)
+			want:          "2006-01-02T15:04:05Z",
+			failureReason: "decoding cbor data tagged with 1 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 1 with float16 value",
+			in:            hex("c1f93c00"), // 1(1.0)
+			want:          "1970-01-01T00:00:01Z",
+			failureReason: "decoding cbor data tagged with 1 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 1 with float32 value",
+			in:            hex("c1fa3f800000"), // 1(1.0)
+			want:          "1970-01-01T00:00:01Z",
+			failureReason: "decoding cbor data tagged with 1 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 1 with float64 value",
+			in:            hex("c1fb3ff0000000000000"), // 1(1.0)
+			want:          "1970-01-01T00:00:01Z",
+			failureReason: "decoding cbor data tagged with 1 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 1 with a four digit year",
+			in:            hex("c11b0000003afff44181"), // 1(253402300801)
+			want:          "10000-01-01T00:00:01Z",
+			failureReason: "decoding cbor data tagged with 1 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 1 with a negative integer value",
+			in:            hex("c120"), // 1(-1)
+			want:          "1969-12-31T23:59:59Z",
+			failureReason: "decoding cbor data tagged with 1 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name:          "tag 1 with a negative float value",
```

Comment:
Please add cases for float32 and float64 too. Negative floats in tag 1 are some of the weirder inputs we might see.

Reply:
Added cases for float32 and float64 negative values.